### DISCUSSION TITLE
Cost Tracker Sensor - select a custom duration

### DISCUSSION THREAD

**Discussion by SteveGBuck** [BottlecapDave/HomeAssistant-OctopusEnergy#709] [Ideas]
I was looking at using the new Cost Tracker Sensor feature to track EV charging costs while on Agile but ideally I'd like to track on a monthly basis rather than daily. Or maybe even session based, daily, monthly, last month, etc... I currently have an automation that achieves this but wondered if it was part of the roadmap to include this capability?

**Comment by BottlecapDave** [comment]
Hello. It wasn't originally part of the roadmap as I _hoped_ there was a built in way in HA to achieve this (like the utility meter but for cost).

Alas, it looks like that's not the case. I want to try and avoid "custom durations" as that can escalate and get complicated quickly.

I've raised https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/714 to accomodate this, but not sure when I'll have the time to look at it.

**Reply by SteveGBuck** [reply]
Don't sweat it ☺️ I already have a working solution. 👍
